src/CanopyContext.js

```diff
@@ -108,11 +110,17 @@ export function CanopyProvider({
   }, []);
 
   useEffect(() => {
-    fetchUserSaplings(saplingURL).then(saplings => {
-      mountSaplingStyles(saplings);
-      mountCurrentSapling(saplings);
-      setUserSaplings(saplings);
-    });
+    if (sessionUser) {
+      fetchUserSaplings(saplingURL).then(saplings => {
+        mountSaplingStyles(saplings);
+        mountCurrentSapling(saplings);
+        setUserSaplings(saplings);
+      });
+    } else {
+      window.$CANOPY.hideCanopy();
```

Comment:
This redirect should really be done by the login sapling, because it would differ if we are using an oauth or saml sapling.

Reply:
Canopy isn't routing to splinter-ui's  login sapling specifically, it's just routing to a sapling that is mapped to `/login` which could be a sapling designed to use oauth or saml for authentication. Given how saplings are currently loaded by canopy, which is to load all the saplings simultaneously, I don't think it's appropriate for the login sapling to be completely responsible for routing users to itself. Something else (like canopy) should be responsible for that. 

Currently, the login sapling is responsible for rendering and routing to itself, but that is what is causing [this bug](https://zenhub.cglcloud.com/app/workspaces/product-core-5d37298b0add7306b30efde2/issues/digitalsupplychain/product-core/2131), both the login sapling and circuits sapling render simultaneously and whichever finishes rendering last "wins". Sometimes that's the circuit sapling which causes the app to skip the login page entirely. The login sapling can't stop the circuit sapling from trying to render because that would mean the login sapling would need to know about other saplings which I don't think is desirable. Some other component, either canopy or some kind of middleware  needs to check if a user is logged in and if they're not, not render the saplings until the user has logged in, and in order for a user to login, that middleware needs to send the user to the correct login sapling. This PR makes canopy responsible for that process, which I think is appropriate since it already knows about users and is responsible for loading saplings.